### ISSUE TITLE
Standardize helm media type for OCI Config Descriptor

### ISSUE THREAD

**Issue by sajayantony** [helm/helm#5286]
Output of `helm version`:

```json
{  "Version":"v3.0+unreleased", 
    "GitCommit":"a32f8ebb377da054d21510671d63729fe6a00f3a", 
    "GitTreeState" :"clean"
}
```
Helm chart push will enable pushing an OCI artifact to the helm registry. This needs to have a standard artifact type that could potentially be used by oci/distribution. 

###  Current Format 
```json
{
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.oci.image.config.v1+json",
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "size": 2
  },
  "layers": [
    {
      "mediaType": "application/vnd.cncf.helm.chart.meta.v1+json",
      "digest": "sha256:5e7c22341d8c8bfaab4dd1acb230a796dbd2d4a57cfb1bc665dbc50671511583",
      "size": 420,
      "annotations": {
        "org.opencontainers.image.title": "chart-meta.json"
      }
    },
    {
      "mediaType": "application/vnd.cncf.helm.chart.content.v1+tar",
      "digest": "sha256:cefbfee797727bffece4f77b9c4c14b041d4f0787fc3edfeedca8e8753b8fd66",
      "size": 18502,
      "annotations": {
        "org.opencontainers.image.title": "chart-content.tgz",
        "sh.helm.chart.name": "wordpress",
        "sh.helm.chart.version": "5.1.2"
      }
    }
  ]
}
```

### Proposal  

OCI manifest allows specifying a config descriptor and the current discussion is to have the proper media type in the config descriptor. 

```json
{
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.cncf.helm.config.v1+json",
...
```

| Type | MediaType | 
|------|------------|
| Config      | application/vnd.cncf.helm.chart.config.v1+json | 
| Metadata | application/vnd.cncf.helm.chart.meta.v1+json |
| Content    | application/vnd.cncf.helm.chart.content.v1+tar | 

/cc @jdolitsky @SteveLasker since this is related to the PR deislabs/oras#52

**Comment by sajayantony** [comment]
Once - https://github.com/deislabs/oras/pull/56 is merged we should enable helm to push a config descriptor with the appropriate mime type.

**Comment by bacongobbler** [comment]
closed via #5719
